File: Java/BinarySearch/Ceil The Floor.java
//https://www.geeksforgeeks.org/problems/ceil-the-floor2802/0
class Solution {
    public int[] getFloorAndCeil(int x, int[] arr) {
        Arrays.sort(arr);  // Sort the array
        
        int floor = findFloor(arr, x);
        int ceil = findCeil(arr, x);
        
        return new int[] { floor, ceil };
    }
    
    // Function to find the floor value
    private int findFloor(int[] arr, int x) {
        int i = 0, j = arr.length - 1;
        if (arr[0] > x) return -1;  // No floor value exists
        
        while (i <= j) {
            int mid = (i + j) / 2;
            if (arr[mid] == x) return arr[mid];
            else if (arr[mid] > x) j = mid - 1;
            else i = mid + 1;
        }
        
        return arr[j];  // Floor value
    }
    
    // Function to find the ceil value
    private int findCeil(int[] arr, int x) {
        int i = 0, j = arr.length - 1;
        if (arr[arr.length - 1] < x) return -1;  // No ceil value exists
        
        while (i <= j) {
            int mid = (i + j) / 2;
            if (arr[mid] == x) return arr[mid];
            else if (arr[mid] > x) j = mid - 1;
            else i = mid + 1;
        }
        
        return arr[i];

File: Python/BinarySearch/Ceil The Floor.py
//https://www.geeksforgeeks.org/problems/ceil-the-floor2802/0
def getFloorAndCeil(self, x: int, arr: list) -> list:
        arr.sort()  # Sort the array
        
        floor = self.findFloor(arr, x)
        ceil = self.findCeil(arr, x)
        
        return [floor, ceil]

    def findFloor(self, arr: list, x: int) -> int:
        i, j = 0, len(arr) - 1
        if arr[0] > x: return -1  # No floor exists
        
        while i <= j:
            mid = (i + j) // 2
            if arr[mid] == x: return arr[mid]
            elif arr[mid] > x: j = mid - 1
            else: i = mid + 1
        
        return arr[j]  # Floor value

    def findCeil(self, arr: list, x: int) -> int:
        i, j = 0, len(arr) - 1
        if arr[j] < x: return -1  # No ceil exists
        
        while i <= j:
            mid = (i + j) // 2
            if arr[mid] == x: return arr[mid]
            elif arr[mid] > x: j = mid - 1
            else: i = mid + 1
        
        return arr[i]
